### DISCUSSION TITLE
rg does not quit after 4 hours

### DISCUSSION THREAD

**Discussion by annazolkieve** [BurntSushi/ripgrep#1661] [General] [answered]
I have 5 GB directory with source code.
Performance comparison: grep vs. rg:
- `grep -r 'qwerty' .` => 14m12s
- `rg 'qwerty'` => rg does not quit after 4 hours => had to interrupt. What can be the reason?

Specs:
```Host OS: Win10 x64.
rg is run under Cygwin:
$ uname -a
CYGWIN_NT-10.0 <NAME> 3.1.6(0.340/5/3) 2020-07-09 08:20 x86_64 Cygwin
$ rg --version
ripgrep 11.0.2 (rev 3de31f7527)
-SIMD -AVX (compiled)
+SIMD -AVX (runtime)
```

**Comment by BurntSushi** [comment, marked as answer]
Impossible to say without a reproduction. Sorry. But the first thing you should do is upgrade. Your version of ripgrep is over 1 year old. You can also try is running ripgrep with `-j1 --trace` and see what the output is and share it if you can. Otherwise, there are two choices:

1. Find a reproduction on a corpus we can both access so that I can reproduce it and debug it.
2. Debug it yourself by inserting `printf` statements in ripgrep source, rebuilding and narrowing down the problem.
